ivy/functional/frontends/torch/tensor.py

```diff
@@ -1011,6 +1011,11 @@ def addr_(self, vec1, vec2, *, beta=1, alpha=1):
     def dot(self, tensor):
         return torch_frontend.dot(self, tensor)
 
+    @with_supported_dtypes({"2.0.1 and below": "float"}, "torch")
+    def bernoulli(self, *, generator=None, out=None):
+        input = torch_frontend.promote_types_of_torch_inputs(self._ivy_array)
```

Comment:
Could you please explain why this line is necessary?

Reply:
The line 
`input=torch_frontend.promote_types_of_torch_inputs(self._ivy_array)` ensures that the input to the bernoulli function is of the correct data type (float) to avoid errors.
Frontend Data Type Promotion Rules:
Reference:https://unify.ai/docs/ivy/overview/deep_dive/ivy_frontends.html#frontend-data-type-promotion-rules

Reply:
`promote_types_of_torch_inputs` takes as input **two** arrays and performs type promotion based on the rules of the native framework. You can take a look at the function signature here: https://github.com/unifyai/ivy/blob/b1f76cafb82889ce95dd84033b2fa67ef2f4ece4/ivy/functional/frontends/torch/__init__.py#L217

Reply:
Okay bro, I will change it..